src/uct/ib/dc/base/dc_iface.c

```diff
@@ -132,6 +132,27 @@ static ucs_status_t uct_dc_iface_dci_connect(uct_dc_iface_t *iface, uct_rc_txqp_
     return UCS_OK;
 }
 
+/* restore dc qp from error to rts state */
+ucs_status_t uct_dc_iface_dci_reconnect(uct_dc_iface_t *iface,
+                                        uct_rc_txqp_t *dci)
+{
+    struct ibv_exp_qp_attr attr;
+
+    memset(&attr, 0, sizeof(attr));
+    attr.qp_state        = IBV_QPS_RESET;
+    attr.pkey_index      = 0;
```

Comment:
you already "zero" everything with memset(), not sure what is the purpose of the following lines